File: CPP/ByRound/0587/0587_A_Duff_and_Weight_Lifting.cpp
/*
 * Problem URL : https://codeforces.com/contest/587/problem/A
 * Submit Date : 2025-09-04
 */

#include<stdio.h>
int a[2000005],ans=0;
 
int main()
{
    int n,x;
    scanf("%d",&n);
    for(int i=1;i<=n;i++)
    {
        scanf("%d",&x);
        a[x]++;
    }
 
    for(int i=0;i<=2000000;i++)
    {
        a[i+1]+=a[i]>>1;
        if(a[i]&1) ans++;
    }
    printf("%d\n",ans);
}

File: Python/ByTier/A/0587_A_Duff_and_Weight_Lifting.py
# Contest 587, Problem A: Duff and Weight Lifting
# URL: https://codeforces.com/contest/587/problem/A

from sys import stdin

input, Max = lambda: stdin.buffer.readline().decode(), 10**6 + 100
n, mem, ans = int(input()), [0] * Max, 0
for i in [int(x) for x in input().split()]:
    mem[i] += 1
for i in range(Max - 1):
    mem[i + 1] += mem[i] >> 1
    ans += mem[i] & 1
print(ans)
